unstable test: TestGetCgroupCPU

## Bug Report

Please answer these questions before submitting your issue. Thanks!
```shell
=== RUN   TestGetCgroupCPU
    cgroup_cpu_test.go:45: 
        	Error Trace:	/home/jenkins/.tidb/tmp/5b8829669b2a941b840f9eb369e2b54f/sandbox/processwrapper-sandbox/2105/execroot/__main__/bazel-out/k8-fastbuild/bin/util/cgroup/cgroup_test_/cgroup_test.runfiles/__main__/util/cgroup/cgroup_cpu_test.go:45
        	Error:      	Received unexpected error:
        	            	no cpu controller detected
        	            	github.com/pingcap/tidb/util/cgroup.getCgroupCPU
        	            		util/cgroup/cgroup_cpu.go:33
        	            	github.com/pingcap/tidb/util/cgroup.GetCgroupCPU
        	            		util/cgroup/cgroup_cpu_linux.go:28
        	            	github.com/pingcap/tidb/util/cgroup.TestGetCgroupCPU
        	            		util/cgroup/cgroup_cpu_test.go:44
        	            	testing.tRunner
        	            		GOROOT/src/testing/testing.go:1446
        	            	runtime.goexit
        	            		GOROOT/src/runtime/asm_amd64.s:1594
        	Test:       	TestGetCgroupCPU
--- FAIL: TestGetCgroupCPU (0.00s)

```
link  https://ci.pingcap.net/blue/organizations/jenkins/tidb-unit-test-hourly/detail/tidb-unit-test-hourly/13322/tests
### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/util/cgroup/cgroup_cpu.go
+++ b/util/cgroup/cgroup_cpu.go
@@ -25,7 +25,16 @@ var noCPUControllerDetected = errors.New("no cpu controller detected")
 
 // Helper function for getCgroupCPU. Root is always "/", except in tests.
 func getCgroupCPU(root string) (CPUUsage, error) {
-	path, err := detectControlPath(filepath.Join(root, procPathCGroup), "cpu,cpuacct")
+	cu, err := getCgroupCPUInternal(root, "cpu,cpuacct")
+	if err == nil {
+		return cu, nil
+	}
+	return getCgroupCPUInternal(root, "cpuacct,cpu")
+}
+
+// getCgroupCPUInternal is to deal with different control keyword
+func getCgroupCPUInternal(root, controlKeyword string) (CPUUsage, error) {
+	path, err := detectControlPath(filepath.Join(root, procPathCGroup), controlKeyword)
 	if err != nil {
 		return CPUUsage{}, err
 	}
@@ -35,7 +44,7 @@ func getCgroupCPU(root string) (CPUUsage, error) {
 		return CPUUsage{}, noCPUControllerDetected
 	}
 
-	mount, ver, err := getCgroupDetails(filepath.Join(root, procPathMountInfo), path, "cpu,cpuacct")
+	mount, ver, err := getCgroupDetails(filepath.Join(root, procPathMountInfo), path, controlKeyword)
 	if err != nil {
 		return CPUUsage{}, err
 	}

--- a/util/cgroup/cgroup_cpu.go
+++ b/util/cgroup/cgroup_cpu.go
@@ -21,6 +21,8 @@ import (
 	"github.com/pingcap/errors"
 )
 
+var noCPUControllerDetected = errors.New("no cpu controller detected")
+
 // Helper function for getCgroupCPU. Root is always "/", except in tests.
 func getCgroupCPU(root string) (CPUUsage, error) {
 	path, err := detectControlPath(filepath.Join(root, procPathCGroup), "cpu,cpuacct")
@@ -30,7 +32,7 @@ func getCgroupCPU(root string) (CPUUsage, error) {
 
 	// No CPU controller detected
 	if path == "" {
-		return CPUUsage{}, errors.New("no cpu controller detected")
+		return CPUUsage{}, noCPUControllerDetected
 	}
 
 	mount, ver, err := getCgroupDetails(filepath.Join(root, procPathMountInfo), path, "cpu,cpuacct")

--- a/util/cgroup/cgroup_cpu.go
+++ b/util/cgroup/cgroup_cpu.go
@@ -21,6 +21,8 @@ import (
 	"github.com/pingcap/errors"
 )
 
+const noCPUControllerDetected = errors.New("no cpu controller detected")
+
 // Helper function for getCgroupCPU. Root is always "/", except in tests.
 func getCgroupCPU(root string) (CPUUsage, error) {
 	path, err := detectControlPath(filepath.Join(root, procPathCGroup), "cpu,cpuacct")
@@ -30,7 +32,7 @@ func getCgroupCPU(root string) (CPUUsage, error) {
 
 	// No CPU controller detected
 	if path == "" {
-		return CPUUsage{}, errors.New("no cpu controller detected")
+		return CPUUsage{}, noCPUControllerDetected
 	}
 
 	mount, ver, err := getCgroupDetails(filepath.Join(root, procPathMountInfo), path, "cpu,cpuacct")

--- a/util/cgroup/cgroup_cpu.go
+++ b/util/cgroup/cgroup_cpu.go
@@ -21,30 +21,19 @@ import (
 	"github.com/pingcap/errors"
 )
 
-var noCPUControllerDetected = errors.New("no cpu controller detected")
-
 // Helper function for getCgroupCPU. Root is always "/", except in tests.
 func getCgroupCPU(root string) (CPUUsage, error) {
-	cu, err := getCgroupCPUInternal(root, "cpu,cpuacct")
-	if err == nil {
-		return cu, nil
-	}
-	return getCgroupCPUInternal(root, "cpuacct,cpu")
-}
-
-// getCgroupCPUInternal is to deal with different control keyword
-func getCgroupCPUInternal(root, controlKeyword string) (CPUUsage, error) {
-	path, err := detectControlPath(filepath.Join(root, procPathCGroup), controlKeyword)
+	path, err := detectControlPath(filepath.Join(root, procPathCGroup), "cpu,cpuacct")
 	if err != nil {
 		return CPUUsage{}, err
 	}
 
 	// No CPU controller detected
 	if path == "" {
-		return CPUUsage{}, noCPUControllerDetected
+		return CPUUsage{}, errors.New("no cpu controller detected")
 	}
 
-	mount, ver, err := getCgroupDetails(filepath.Join(root, procPathMountInfo), path, controlKeyword)
+	mount, ver, err := getCgroupDetails(filepath.Join(root, procPathMountInfo), path, "cpu,cpuacct")
 	if err != nil {
 		return CPUUsage{}, err
 	}

--- a/util/cgroup/cgroup_cpu.go
+++ b/util/cgroup/cgroup_cpu.go
@@ -21,7 +21,7 @@ import (
 	"github.com/pingcap/errors"
 )
 
-const noCPUControllerDetected = errors.New("no cpu controller detected")
+var noCPUControllerDetected = errors.New("no cpu controller detected")
 
 // Helper function for getCgroupCPU. Root is always "/", except in tests.
 func getCgroupCPU(root string) (CPUUsage, error) {

--- a/util/cgroup/cgroup_cpu.go
+++ b/util/cgroup/cgroup_cpu.go
@@ -21,7 +21,7 @@ import (
 	"github.com/pingcap/errors"
 )
 
-var noCPUControllerDetected = errors.New("no cpu controller detected")
+var errNoCPUControllerDetected = errors.New("no cpu controller detected")
 
 // Helper function for getCgroupCPU. Root is always "/", except in tests.
 func getCgroupCPU(root string) (CPUUsage, error) {
@@ -32,7 +32,7 @@ func getCgroupCPU(root string) (CPUUsage, error) {
 
 	// No CPU controller detected
 	if path == "" {
-		return CPUUsage{}, noCPUControllerDetected
+		return CPUUsage{}, errNoCPUControllerDetected
 	}
 
 	mount, ver, err := getCgroupDetails(filepath.Join(root, procPathMountInfo), path, "cpu,cpuacct")

--- a/util/cgroup/cgroup_cpu_test.go
+++ b/util/cgroup/cgroup_cpu_test.go
@@ -42,7 +42,7 @@ func TestGetCgroupCPU(t *testing.T) {
 		}()
 	}
 	cpu, err := GetCgroupCPU()
-	if err == noCPUControllerDetected {
+	if err == errNoCPUControllerDetected {
 		require.False(t, InContainer(), "Please check linux version. This is related to cgroup compatibility.")
 	} else {
 		require.NoError(t, err)

--- a/util/cgroup/cgroup_cpu_test.go
+++ b/util/cgroup/cgroup_cpu_test.go
@@ -43,7 +43,7 @@ func TestGetCgroupCPU(t *testing.T) {
 	}
 	cpu, err := GetCgroupCPU()
 	if err == noCPUControllerDetected {
-		require.false(t, InContainer())
+		require.False(t, InContainer())
 	} else {
 		require.NoError(t, err)
 		require.NotZero(t, cpu.Period)

--- a/util/cgroup/cgroup_cpu_test.go
+++ b/util/cgroup/cgroup_cpu_test.go
@@ -43,7 +43,7 @@ func TestGetCgroupCPU(t *testing.T) {
 	}
 	cpu, err := GetCgroupCPU()
 	if err == errNoCPUControllerDetected {
-		require.False(t, InContainer(), "Please check linux version. This is related to cgroup compatibility.")
+		require.False(t, InContainer(), "Please check linux version > v4.7.x. This is related to cgroup compatibility.")
 	} else {
 		require.NoError(t, err)
 		require.NotZero(t, cpu.Period)

--- a/util/cgroup/cgroup_cpu_test.go
+++ b/util/cgroup/cgroup_cpu_test.go
@@ -41,6 +41,7 @@ func TestGetCgroupCPU(t *testing.T) {
 			}
 		}()
 	}
+	require.True(t, InContainer())
 	cpu, err := GetCgroupCPU()
 	require.NoError(t, err)
 	require.NotZero(t, cpu.Period)

--- a/util/cgroup/cgroup_cpu_test.go
+++ b/util/cgroup/cgroup_cpu_test.go
@@ -41,11 +41,14 @@ func TestGetCgroupCPU(t *testing.T) {
 			}
 		}()
 	}
-	require.True(t, InContainer())
 	cpu, err := GetCgroupCPU()
-	require.NoError(t, err)
-	require.NotZero(t, cpu.Period)
-	require.Less(t, int64(1), cpu.Period)
+	if err == noCPUControllerDetected {
+		require.false(t, InContainer())
+	} else {
+		require.NoError(t, err)
+		require.NotZero(t, cpu.Period)
+		require.Less(t, int64(1), cpu.Period)
+	}
 	close(exit)
 	wg.Wait()
 }

--- a/util/cgroup/cgroup_cpu_test.go
+++ b/util/cgroup/cgroup_cpu_test.go
@@ -43,7 +43,7 @@ func TestGetCgroupCPU(t *testing.T) {
 	}
 	cpu, err := GetCgroupCPU()
 	if err == noCPUControllerDetected {
-		require.False(t, InContainer())
+		require.False(t, InContainer(), "Please check linux version. This is related to cgroup compatibility.")
 	} else {
 		require.NoError(t, err)
 		require.NotZero(t, cpu.Period)

